examples/bluetooth/ble_hid_mouse.py

```diff
@@ -0,0 +1,110 @@
+    else:
+        print("event:", event, data)
+
+
+ble = bluetooth.BLE()
+ble.active(1)
+ble.irq(ble_irq)
+
+UUID = bluetooth.UUID
+
+F_READ = bluetooth.FLAG_READ
+F_WRITE = bluetooth.FLAG_WRITE
+F_READ_WRITE = bluetooth.FLAG_READ | bluetooth.FLAG_WRITE
+F_READ_NOTIFY = bluetooth.FLAG_READ | bluetooth.FLAG_NOTIFY
+
+ATT_F_READ = 0x01
+ATT_F_WRITE = 0x02
+
+hid_service = (
+    UUID(0x1812),  # Human Interface Device
+    (
+        (UUID(0x2A4A), F_READ),  # HID information
+        (UUID(0x2A4B), F_READ),  # HID report map
+        (UUID(0x2A4C), F_WRITE),  # HID control point
+        (UUID(0x2A4D), F_READ_NOTIFY, ((UUID(0x2908), ATT_F_READ),)),  # HID report / reference
+        (UUID(0x2A4E), F_READ_WRITE),  # HID protocol mode
+    ),
+)
+
+# fmt: off
+HID_REPORT_MAP = bytes([
+    0x05, 0x01,     # Usage Page (Generic Desktop)
+    0x09, 0x02,     # Usage (Mouse)
+    0xA1, 0x01,     # Collection (Application)
+    0x09, 0x01,     #     Usage (Pointer)
+    0xA1, 0x00,     #     Collection (Physical)
+    0x85, 0x01,     #         Report ID (1)
+    0x95, 0x03,     #         Report Count (3)
+    0x75, 0x01,     #         Report Size (1)
+    0x05, 0x09,     #         Usage Page (Buttons)
+    0x19, 0x01,     #         Usage Minimum (1)
+    0x29, 0x03,     #         Usage Maximum (3)
+    0x15, 0x00,     #         Logical Minimum (0)
+    0x25, 0x01,     #         Logical Maximum (1)
+    0x81, 0x02,     #         Input(Data, Variable, Absolute); 3 button bits
+    0x95, 0x01,     #         Report Count(1)
+    0x75, 0x05,     #         Report Size(5)
+    0x81, 0x01,     #         Input(Constant);                 5 bit padding
+    0x75, 0x08,     #         Report Size (8)
+    0x95, 0x02,     #         Report Count (3)
+    0x05, 0x01,     #         Usage Page (Generic Desktop)
+    0x09, 0x30,     #         Usage (X)
+    0x09, 0x31,     #         Usage (Y)
+    0x09, 0x38,     #         Usage (Wheel)
+    0x15, 0x81,     #         Logical Minimum (-127)
+    0x25, 0x7F,     #         Logical Maximum (127)
+    0x81, 0x06,     #         Input(Data, Variable, Relative); 3 position bytes (X,Y,Wheel)
+    0xC0,           #     End Collection
+    0xC0,           # End Collection
+])
+# fmt: on
+
+# register services
+ble.config(gap_name="MP-mouse")
+handles = ble.gatts_register_services((hid_service,))
+print(handles)
+h_info, h_hid, _, h_rep, h_d1, h_proto = handles[0]
+
+# set initial data
+ble.gatts_write(h_info, b"\x01\x01\x00\x02")  # HID info: ver=1.1, country=0, flags=normal
+ble.gatts_write(h_hid, HID_REPORT_MAP)  # HID report map
+ble.gatts_write(h_d1, struct.pack("<BB", 1, 1))  # report: id=1, type=input
+ble.gatts_write(h_proto, b"\x01")  # protocol mode: report
+
+# advertise
+adv = (
+    b"\x02\x01\x06"
+    b"\x03\x03\x12\x18"  # complete list of 16-bit service UUIDs: 0x1812
+    b"\x03\x19\xc2\x03"  # appearance: mouse
+    b"\x09\x09MP-mouse"  # complete local name
+)
+conn_handle = None
+ble.gap_advertise(100_000, adv)
```

Comment:
```py
import ble_advertising
_ADV_APPEARANCE_MOUSE = const(962)
adv = ble_advertising.advertising_payload(services=[UUID(0x1812)], appearance=_ADV_APPEARANCE_MOUSE, name="MP-mouse")
```

Reply:
I kind of wanted this example to be raw and just use the `bluetooth` module as-is, without any helpers.  As a way to show exactly what's needed to make a BLE HID, to show there's no complexity hiding anywhere.  IMO that's valuable for learning about BLE (and HID).  That's also why I added mouse and keyboard as separate but very similar files.

So maybe we can have a "raw" example (perhaps just mouse, it's simpler) and then a more "friendly" version which has both keyboard and mouse support (selectable via some option).

And then eventually also a proper (async) HID library, but that's for the future.

Reply:
OK, sounds good!